### DISCUSSION TITLE
Tweak Boost behaviour?

### DISCUSSION THREAD

**Discussion by MichK-tech** [BartoszCichecki/LenovoLegionToolkit#1063] [Ideas]
RyzenAdj has these 2 options:

 --power-saving                        Hidden options to improve power efficiency (is set when AC unplugged): behavior depends on CPU generation, Device and Manufacture
    --max-performance                     Hidden options to improve performance (is set when AC plugged in): behavior depends on CPU generation, Device and Manufacture

Is it possible do implement them in LLT?

**Comment by BartoszCichecki** [comment]
I don't know what these options are.

**Reply by MichK-tech** [reply]
**--power-saving:** CPU Boosts slower (like it does on default while on battery)
**--max-performance:** CPU Boosts faster (like it does on default while plugged in)

Personally I liked the power-saving very much when plugged in on my 5800H, because the boost frequency is only reached after a longer peroid of cpu-load. (which results in performance loss but better temperature and therefor lower spikes in fan speed). I think this setting would be helpful on high wattage cpus like (7945HX) - short load spikes don't need 5.4GHz, but longer loads (like rendering) should be done with boost speeds. On the other hand you can unleash additional performance while on battery: https://www.reddit.com/r/Amd/comments/mkinei/unleash_ryzen_mobile_boost_delay_on_battery/

_(another idea - I'm afraid to ask - did you already thought about implementing Curve Optimizer options for 7945HX like UXTU V1.1.0 Beta 5 has)_

**Reply by BartoszCichecki** [reply]
Ok, so that basically control CPU Boost settings. Why would it need to included if it can be changed in Windows power plan settings? And in other software? Same with the curve thing. What's the point of spending time on re-implementing it in LLT if there already is UXTU doing it well?

**Reply by MichK-tech** [reply]
At least on 5800H it does not only affect boost settings. Also the idle clock speeds are lower (like - I think - they are on any Ryzen, while on battery)  Basically it these settings allow battery behaviour (lower idle-clocks, leisurely boost) when plugged-in and vice-versa. I don't think that this can achived with windows power plan settings. and for the curve thing - uxtu only allows it with an older version. Also it would not be necessary to use different tweaking tools. _(but the boost thing is much more interesting)_

**Reply by BartoszCichecki** [reply]
I am not an AMD expert, in fact I don't even have an AMD laptop, but to me it sounds like a general AMD feature. I don't think it fits into "replacing Vantage" vision this project follows. RyzenAdj can be integrated in LLT actions (if needed) using the Run step. Regarding, curve optimizer, I have no idea why it no longer is part of newer versions of UXTU, it might be worth asking them instead.

The problem I see here is that stuff like this quickly grows. LLT needs to be lightweight. Now you are asking for RyzenAdj/UXTU features in LLT, next someone will ask for Intel XTU stuff, next will be Afterburner and so on and LLT will grow to be bigger and slower and worse than Vantage - and we don't want that.
